codegen/templates/dependency_struct.tmpl

```diff
@@ -0,0 +1,24 @@
+{{$instance := . -}}
+package {{$instance.PackageInfo.PackageName}}
+
+import (
+	{{range $classType, $moduleInstances := $instance.ResolvedDependencies -}}
+	{{range $idx, $moduleInstance := $moduleInstances -}}
+	{{$moduleInstance.PackageInfo.ImportPackageAlias}} "{{$moduleInstance.PackageInfo.ImportPackagePath}}"
+	{{end -}}
+	{{end -}}
+)
+
+type {{$instance.PackageInfo.ExportType}}Deps struct {
+	{{range $classType, $moduleInstances := $instance.ResolvedDependencies -}}
+	{{$classType | pascal}} {{$instance.PackageInfo.ExportType}}{{$classType | pascal}}Deps
```

Comment:
If we deviate from the class type, that deviation needs to be encoded somewhere. I'd rather flatten the dependency list than try to somehow figure out that the plural of Client is Clients.